src/modm/platform/core/e31/module.lb

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Benjamin Weps
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":platform:core"
+    module.description = "SiFive E31 Core"
+
+def common_memories(env):
+    """
+    Computes memory properties:
+
+      - `memories`: unfiltered memory regions
+      - `regions`: memory region names
+
+    :returns: dictionary of memory properties
+    """
+    device = env[":target"]
+    memories = listify(device.get_driver("core")["memory"])
+
+    # Convert from string to int and add offsets
+    flash_offset = env.get(":platform:core:linkerscript.flash_offset", 0)
+    for m in memories:
+        if m["name"] == "flash":
+            m["start"] = int(m["start"], 0) + flash_offset
+            m["size"] = int(m["size"], 0) - flash_offset
+        else:
+            m["start"] = int(m["start"], 0)
+            m["size"] = int(m["size"], 0)
+
+    properties = {
+        "memories": memories,
+        "regions": [m["name"] for m in memories],
+    }
+    return properties
+
+
+def prepare(module, options):
+    if not options[":target"].has_driver("core:e31"):
+        return False
+
+    module.depends(
+            ":architecture:assert",
+            ":architecture:memory"
+        )
+
+    module.add_option(
+    EnumerationOption(
+        name="allocator",
+        description=FileReader("option/allocator.md"),
```

Comment:
Please link to the Cortex-M description here please and make it less Cortex-M specific (things like replace `arm-none-eabi` -> `toolchain`): `"../cortex/option/allocator.md"`